wordpress-mobile/WordPress-Android
# ViewBinding: Replace synthetic accessors in LayoutPreviewFragment and children

Parent #14845 

This PR is part of a larger effort to replace synthetic accessors within WPAndroid.

ViewHolders updated
- CategoryViewHolder
- LayoutViewHolder

Fragments
- LayoutPreviewFragment
- BlockLayoutFragment
- DesignPreviewFragment

In addition to the view binding updates, two new abstract methods were added `LayoutPreviewFragment` for getting button text and setting the view model. The `onAttach` method could not be abstracted out because the children fragment are injected via dagger.

Note: I went with .the following because I find this cleaner than `?:` - neither is wrong, to me, is reads more easily. :) 

```
    private var _binding: LayoutPickerPreviewFragmentBinding? = null    
    private val binding get() = _binding!!
```

**To test:**
- Launch the app
- Navigate to My Site -> FAB -> Site Page
- (1) Note that the category bar along the top works correctly (select/unselect)
-  Select a page layout (you will see the check on the page layout)
- Tap the preview button
- (2) Note that the preview shows as expected
- Navigate back to My Site
- Tap the site chooser
- Select + in the upper right hand corner and tap "Create WordPress.com site"
- (3) Note that the category bar along the top works correctly
- Select a page layout (you will see the check on the page layout)
- Tap the preview button
- (4) Note that the preview shows as expected

## Regression Notes
1. Potential unintended areas of impact
The site page and new site page layouts do not work as expected.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested both of those processes.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
